src/components/CRCMaker.js

```diff
@@ -243,19 +247,65 @@ export default class CRCMaker extends Component {
     });
   }
 
-  // generatePDF () {
-  //   window.scrollTo(0, 0);
-  //   var cards = document.getElementById('cards');
-
-  //   html2canvas(cards).then((canvas) => {
-  //     console.log(canvas);
-  //     var img = canvas.toDataUrl('image/png');
-
-  //     var doc = new jsPDF();
-  //     doc.addImage(img, 'JPEG', 20, 20);
-  //     doc.save('test_img.pdf');
-  //   });
-  // }
+  generatePDF () {
+    window.scrollTo(0, 0);
+    var doc = new PDFDocument()
+    var stream = doc.pipe(blobStream());
+    
+    var cursorX = 20;
+    var cursorY = 20;
+
+    var threshold = (72 * 11) - 20
+
+    this.state.cards.map((data, i) => {
+      //get height of card
+      var maxItems = Math.max(data.responsibilities.length, data.collaborators.length);
+      var height = maxItems*15 + 100;
+      var width = 572
+
+      if(cursorY + height > threshold){
+        doc.addPage();
+        cursorY = 20;
+      } 
+
+      let type = '';
+      if (data.type == CardTypes.ABSTRACT) {
+        type = 'Abstract';
+      } else if (data.type == CardTypes.INTERFACE) {
+        type = 'Interface';
+      }
+      var superclasses = data.superclasses;
+      var name = data.name;
+      var subclasses = data.subclasses;
+
+      doc.fontSize(15);
+      doc.text(type, cursorX + 3, cursorY + 10);
```

Comment:
There's a whole bunch of magic numbers going on here. Could we have comments or something that explain what they're for?